src/TreeHouse/Cache/Driver/FileDriver.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace TreeHouse\Cache\Driver;
+
+class FileDriver implements DriverInterface
+{
+    /**
+     * @var string
+     */
+    private $cacheDir;
+
+    /**
+     * @param string $cacheDir
+     */
+    public function __construct($cacheDir)
+    {
+        if (!is_dir($cacheDir) || !is_readable($cacheDir)) {
+            throw new \LogicException(sprintf('Cache dir does not exist or is not readable: "%s"', $cacheDir));
+        }
+
+        $this->cacheDir = $cacheDir;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function has($key)
+    {
+        $file = $this->getCacheFile($key);
+
+        return file_exists($file);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function get($key)
+    {
+        $file = $this->getCacheFile($key);
+
+        if (!file_exists($file)) {
+            return false;
+        }
+
+        if (!is_readable($file)) {
+            throw new \RuntimeException(sprintf('Cache file is unreadable: "%s"', $file));
+        }
+
+        if (false === $contents = file_get_contents($file)) {
```

Comment:
Voor grotere bestanden zou dat wenselijk zijn ja, hier lijkt me dat niet het geval. Bovendien is het niet consistent met de overige drivers (die ook de hele value terug geven).